src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1842,90 +1842,71 @@ private static void GetAccessibilityAndModifiers(SyntaxTokenList modifierList, o
                         break;
 
                     case SyntaxKind.PrivateKeyword:
-                        if (accessibility == Accessibility.Protected)
-                        {
-                            accessibility = Accessibility.ProtectedAndInternal;
-                        }
-                        else
-                        {
-                            accessibility = Accessibility.Private;
-                        }
+                        accessibility = accessibility == Accessibility.Protected
+                            ? Accessibility.ProtectedAndInternal
+                            : Accessibility.Private;
                         break;
 
                     case SyntaxKind.InternalKeyword:
-                        if (accessibility == Accessibility.Protected)
-                        {
-                            accessibility = Accessibility.ProtectedOrInternal;
-                        }
-                        else
-                        {
-                            accessibility = Accessibility.Internal;
-                        }
-
+                        accessibility = accessibility == Accessibility.Protected
+                            ? Accessibility.ProtectedOrInternal
+                            : Accessibility.Internal;
                         break;
 
                     case SyntaxKind.ProtectedKeyword:
-                        if (accessibility == Accessibility.Private)
-                        {
-                            accessibility = Accessibility.ProtectedAndInternal;
-                        }
-                        else if (accessibility == Accessibility.Internal)
-                        {
-                            accessibility = Accessibility.ProtectedOrInternal;
-                        }
-
-                        {
-                            accessibility = Accessibility.Protected;
-                        }
-
+                        accessibility = accessibility == Accessibility.Private
+                            ? Accessibility.ProtectedAndInternal
+                            : accessibility == Accessibility.Internal
+                                ? Accessibility.ProtectedOrInternal
+                                : Accessibility.Protected;
```

Comment:
💡 `switch` expressions finally offer a clean alternative to multiply-nested ternary